### PR TITLE
ENT-10182: Augmentable TLS settings for inbound connections (body server control)

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by craigcomstock** [comment]
@nickanderson this PR will fix your broken tests: https://github.com/cfengine/masterfiles/pull/2608

File: controls/cf_serverd.cf [hunk 1]
```diff
@@ -44,6 +31,26 @@ body server control
...
 
+    control_server_allowtlsversion_defined::
+
+      # Minimum required version of TLS. Set to "1.0" if you need clients
+      # running CFEngine in a version lower than 3.7.0 to connect.
+      #
+      # Example:
+      # allowtlsversion => "1.0";
+
+      allowtlsversion => "$(default:def.control_server_allowtlsversion)";
```
Comment (craigcomstock) [inline controls/cf_serverd.cf:42]: seeing this in isolation and trying to think as a new user, I wonder if `def.` wouldn't be better as `augment.` or even `config.`. As well as a mention when we have these in places to give someone a clue where to look next: aka Augments in docs.cfengine.com or MPF.md.

File: standalone_self_upgrade.cf.in [hunk 2]
```diff
@@ -764,6 +780,13 @@ body common control
 {
       version => "CFEngine Standalone Self Upgrade @VERSION@";
 
+
+    control_common_tls_min_version_defined::
```
Comment (craigcomstock) [inline standalone_self_upgrade.cf.in:784]: maybe a "see controls/cf_serverd.cf" in here and in update.cf?

File: controls/cf_serverd.cf (hunk 1 above)
Comment (nickanderson) [reply controls/cf_serverd.cf:42]: Well, there is a lot of history around `def`.

* Prior to 3.18.0 Augments would ONLY target `default:def`.
* The augments_control class that bounces components based on data change does so by targeting `default:def.control_<component>.*` and figures out which components need re-started based on that. So, even though we can now leverage variables outside `default:def` I have not strayed for anything that ends up being used in a `control` body.


This PR contained changes to `MPF.md` which is rendered in the docs to highlight how to configure it. Were you looking for something different here?

File: standalone_self_upgrade.cf.in (hunk 2 above)
Comment (nickanderson) [reply standalone_self_upgrade.cf.in:784]: I added cross comments about the other body to find related settings.

**Review by craigcomstock — APPROVED** [review]

**Pull request merged**